modules/ballerina-core/src/main/java/org/ballerinalang/model/util/JsonNode.java

```diff
@@ -0,0 +1,398 @@
+        if (this.fields == null) {
+            this.fields = new LinkedHashMap<>();
+        }
+        this.fields.put(name, new JsonNode(booleanValue));
+    }
+    
+    public void set(String name, JsonNode jsonNode) {
+        if (this.fields == null) {
+            this.fields = new LinkedHashMap<>();
+        }
+        if (jsonNode != null) {
+            this.fields.put(name, jsonNode);
+        } else {
+            this.fields.put(name, new JsonNode(Type.NULL));
+        }
+    }
+    
+    public void set(int index, JsonNode jsonNode) {
+        if (this.arrayElements == null) {
+            throw new BallerinaException("JSON array is empty to set values");
+        }
+        this.arrayElements.set(index, jsonNode);
+    }
+    
+    public boolean has(String name) {
+        return this.get(name) != null;
+    }
+    
+    public JsonNode get(String name) {
+        if (this.fields != null) {
+            return this.fields.get(name);
+        } else {
+            return null;
+        }
+    }
+    
+    public JsonNode get(int index) {
+        if (this.arrayElements != null) {
+            return this.arrayElements.get(index);
+        } else {
+            return null;
+        }
+    }
+    
+    public String asText() {
+        switch (this.type) {
+        case BOOLEAN:
+            return Boolean.toString(this.booleanValue());
+        case DOUBLE:
+            return Double.toString(this.doubleValue());
+        case LONG:
+            return Long.toString(this.longValue());
+        case NULL:
+            return "null";
+        case STRING:
+            return this.stringValue();
+        default:
+            return "";
+        }
+    }
+    
+    public Iterator<String> fieldNames() {
+        if (this.fields != null) {
+            return this.fields.keySet().iterator();
+        } else {
+            return new HashSet<String>(0).iterator();
+        }
+    }
+    
+    public Iterator<Entry<String, JsonNode>> fields() {
+        if (this.fields != null) {
+            return this.fields.entrySet().iterator();
+        } else {
+            return new LinkedHashMap<String, JsonNode>(0).entrySet().iterator();
+        }
+    }
+    
+    public Iterator<JsonNode> elements() {
+        if (this.arrayElements != null) {
+            return this.arrayElements.iterator();
+        } else {
+            return new ArrayList<JsonNode>(0).iterator();
+        }
+    }
+    
+    public void remove(String fieldName) {
+        if (this.fields != null) {
+            this.fields.remove(fieldName);
+        }
+    }
+    
+    public String toString() {
+        ByteArrayOutputStream byteOut = new ByteArrayOutputStream();
+        JsonGenerator gen = new JsonGenerator(byteOut);
+        try {
+            this.serialize(gen);
+            gen.flush();
+        } catch (IOException e) {
+            throw new BallerinaException("Error in JSON toString: " + e.getMessage(), e);
```

Comment:
Exposing Java level error messages to the Ballerina developer.  Can we reword the error message? Possibly be removing the term "toString"

Reply:
Yep, true, fixed in the latest commit.